### DISCUSSION TITLE
关于样式隔离、全局变量、全局事件的隔离问题

### DISCUSSION THREAD

**Discussion by JSjiumi** [empjs/emp#191] [Ideas]
我已经按照教程建立了emp的demo，并且成功运行起来了，应用间引用和交互都正常。
现在有一个疑问，感觉emp更倾向于共享库的概念，以组件的方式使用共享的应用。那么，作为微前端考虑，应用彼此间的样式和全局事件的隔离是怎么处理的呢？在github和google上并没有找到相关的介绍和说明。

**Comment by ckken** [comment]
隔离场景可以描述下

**Comment by axlroseart** [comment]
@ckken 你好，关于EMP模式下的应用之间JS隔离我有些疑惑，想请教一下：

- 某子应用和remote引入的组件里面都有注册同一个'blur'事件，比如：

// event listener in host app:
window.addEventListener('blur', () => {
  console.log('blur event triggered in host app');
});

// event listener in remote app:
window.addEventListener('blur', () => {
  console.log('blur event triggered in remote app');
});

- 某子应用和remote引入的组件中都有挂在window下的全局变量，且变量名相同

这两种情况下，事件的执行和全局变量的访问结果会是怎样的？
补充一下，host和remote应用是嵌套运行的，也就是两个应用同时加载，运行在同一个页面上
